src/core_landice/mode_forward/mpas_li_calving.F

```diff
@@ -1181,10 +1187,11 @@ subroutine eigencalving(domain, err)
 
          calvingVelocity(:) = 0.0_RKIND
          ! First calculate the front retreat rate (Levermann eq. 1)
-         calvingVelocity(:) = eigencalvingParameter(:) * max(0.0_RKIND, eMax(:)) * max(0.0_RKIND, eMin(:)) & ! m/s
```

Comment:
> @tongzhangice , thanks for getting all these changes in a clear and concise form. I've marked a number changes to be made, and asked some questions. Let me know if anything is not clear.
> 
> I have a few more general questions:
> 
> 1. Can you explain what the purpose of the restore_damage routine is?
> 2. I did not see anywhere the possibility for damage healing.  How are you testing with that on or off?
> 3. I'm confused by your method of initializing damage at the grounding line.  It looks like it's coming from the damage downstream, which sounds like a circular argument.
> 4. In Registry, it indicates that it can be set up for either calving to occur when damage exceeds some value or for calving velocity to be proportional to the damage value.  But in the code I only see the latter and not the former.  Is it still possible to make ice completely calve when damage surpasses a threshold?
> 
> A couple less important issues:
> 5. Eventually we should make a package for the damage variables so they are only allocated and input from file if the damage scheme is on. I can help set that up.
> 6. I think it would make more sense to move all the viscosity and stress calculations to a new subroutine that is independent of the damage code.

@matthewhoffman I've addressed most of your comments below. Replies to your points:
1. the restore_damage function is to decide if we want to allow damage to heal. If we restore the damage in the run, we do not allow damage to heal, i.e., the local damage will not be decreased by advection.
2. see 1)
3. the reason is we don't want GL's downstream neighboring cells to heal during the run, because the damage for grounded ice is always zero in the damage model.
4. Right. I forgot to put the threshold method in it. It is now added.

Reply:
A new subroutine 'specified_calving_velocity' is added, and the Registry is also updated.